acceptance/builder_test.go

```diff
@@ -0,0 +1,367 @@
+				})
+			})
+
+		})
+	})
+
+	when("PlaceHolders need to replace with defaults", func() {
+
+		// .../cmd/lifecycle/builder.go#Args
+		when("groupPath is equals to PlaceHolder groupPath", func() {
+			it("will replace placeholder with default groupPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=<layers>/group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to read buildpack group: open /layers/group.toml"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#Args
+		when("planPath is equals to PlaceHolder planPath", func() {
+			it("will replace placeholder with default planPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=<layers>/plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan: open /layers/plan.toml"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+	})
+
+	/// .../cmd/lifecycle/builder.go#build
+	when("Builder args are successfully transmitted to in build script", func() {
+		when("CNB_APP_DIR changed", func() {
+			it("CNB_APP_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_APP_DIR=/env_folders/different_cnb_app_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "CNB_APP_DIR: /env_folders/different_cnb_app_dir_from_env"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_BUILDPACKS_DIR changed", func() {
+			it("CNB_BUILDPACKS_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_BUILDPACKS_DIR=/env_folders/different_buildpack_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
```

Comment:
Similar to other comment (https://github.com/buildpacks/lifecycle/pull/648/files#r669952363) we could pass a real directory, and the buildpack found there could print "Hello from other directory" which we could assert on.

Reply:
See https://github.com/buildpacks/lifecycle/pull/648/files#r669954888

Reply:
we are already printing CNB_BUILDPACK_DIR: env variable and asserting this one, if u want i can add "Hello from other directory" string, i also update build script for this buildpack. please check again.

Reply:
What you did works!